source/code/plugins/filter_vmware_logs.rb

```diff
@@ -0,0 +1,54 @@
+module Fluent
+  class VmwareSyslogFilter < Filter
+
+    Fluent::Plugin.register_filter('filter_vmware_logs', self)
+
+    def initialize
+      super
+    end
+
+    def configure(conf)
+      super
+    end
+
+    def start
+      super
+    end
+
+    def shutdown
+      super
+    end
+
+    def filter(tag, time, record)
+      # Co-relation fields
+      record['Computer'] = record['HostName']
+      record['ResourceName'] = 'VMware'
+      record['ResourceLocation'] = 'VMware'
+      record['ResourceType'] = 'Hypervisor'
+      record['ResourceId'] = record['HostName']
+
+      begin
+        # Regex for Device. Example string: naa.60a9800041764b6c463f43786855a3t2
+        record['Device'] = record['SyslogMessage'].to_s.match(/naa\.[a-z0-9]{32}/).to_s
```

Comment:
I would recommend you extract the logic into a parser plugin that the syslog plugin can apply.
http://docs.fluentd.org/articles/parser-plugin-overview#

You can use the built-in record_transform plugin for adding, removing or modifying fields.
http://docs.fluentd.org/articles/filter_record_transformer